src/boot/pe.c

```diff
@@ -441,17 +469,36 @@ EFI_STATUS pe_kernel_info(const void *base, uint32_t *ret_compat_address, size_t
         if (!verify_pe(dos, pe, /* allow_compatibility= */ true))
                 return EFI_LOAD_ERROR;
 
+        uint64_t image_base;
+        if (pe->OptionalHeader.Magic == OPTHDR32_MAGIC)
+                image_base = pe->OptionalHeader.ImageBase32;
+        else
+                image_base = pe->OptionalHeader.ImageBase64;
+
+        const PeImageDataDirectory *data_directory;
+        if (pe->OptionalHeader.Magic == OPTHDR32_MAGIC)
+                data_directory = pe->OptionalHeader.DataDirectory32;
+        else
+                data_directory = pe->OptionalHeader.DataDirectory64;
```

Comment:
can you merge these two if expressions. maybe even turn into a switch expression. and also please add an asser_not_reached() call if neither OPTHDR32_MAGIC nor OPTHDR64_MAGIC are seen here. (this must hold, after all verify_pe() already checked that, hence an assert here is fine)

Reply:
I had the move that part to another function, because we are not supposed to check relocations in pe_kernel_info when called from systemd-boot. But I will change to a switch so I can add the assert_not_reached.